scripts/legislation/africa/malawi_english.sh

```diff
@@ -0,0 +1,236 @@
+#!/bin/bash
+
+function remove_all_text_before_first_header {
+  sed -n '/^ ENACTED by the Parliament of Malawi as follows/,$p' | \
+    sed -n '/^ PART I -/,$p'
+}
+
+function prefix_article_numbers_with_article_literal {
+  sed -E 's/^([0-9]+\.)/Article \1/'
+}
+
+function replace_parentheses_around_article_delimiters_with_square_brackets {
+  sed -E 's/^\(([A-Za-z0-9]+)\)/[\1]/' 
+}
+
+function amend_errors_in_headers {
+  sed -E 's/PART/\n\nPART/g' | \
+    sed -E 's/^(PART .*)\[•\]/\1-/' | \
+    sed -n '/PART I /,$p' | \
```

Comment:
This line seems redundant (`remove_all_text_before_first_header`). Am I mistaken?

Reply:
Unlike with the Uganda legislation, `sed -E 's/([^^])PART/\1\n\nPART/g' | \` is unable to remove all of the space before PART I which `sed -n '/PART I /,$p' | \` does successfully. This is likely due to the additional spaces before PART I, thus I believe these 3 lines are necessary for the proper formatting.

Reply:
The purpose of the command `sed -E 's/([^^])PART/\1\n\nPART/g'` is to add two newline characters before each instance of `PART` that does not occur at the beginning of a line. The purpose of the command `sed -E 's/PART/\n\nPART/g'` is basically the same, although it will match any instance of `PART`, regardless of where it occurs in a given line. Importantly, neither of these commands is specifically designed to remove leading spaces. The fact that they separate leading spaces from the word `PART` by newline characters is incidental to their purpose. As such, if you simply want to remove spaces between the beginning of a line and the word `PART`, then you may want to craft a command for that particular purpose.

All of that said, the command on line 17 is "removing" the spaces before `PART`, not the command on line 19, so I still believe the command on line 19 is redundant. Furthermore, in the event that `PART` is preceded by a space on a given line, then the command on line 17 and the command `sed -E 's/([^^])PART/\1\n\nPART/g'` should be interchangeable. Are you sure that the latter will not insert newline characters before `PART I` in this case?

Reply:
From my observation, both `sed -E 's/([^^])PART/\1\n\nPART/g'` and `'s/PART/\n\nPART/g'` will insert two newlines before all PART titles, including the first PART title which is "PART I - PRELIMINARY". 

Line 19, `sed -n '/PART I /,$p' | \` operates to remove those 2 newlines before the first PART title so that the preprocessed legislation does not start with 2 newlines. So to me, line 19 seems to be necessary unless there is a command that can insert 2 newlines before PART titles but not before the first PART title. Not sure if there's something that I'm missing.

Reply:
Oh, I see what you are saying. To solve that problem, you can delete line 19 and add the following before line 17:

```
sed -E 's/^ +PART/PART/'
```

That will remove the leading spaces so the command currently on line 17 will not add newlines before `PART I ...`